src/js/angular/clustermanagement/controllers.js

```diff
@@ -218,55 +219,27 @@ function ClusterManagementCtrl($scope, $http, $q, toastr, $repositories, $modal,
 
     $scope.getCurrentNodeStatus = () => {
         return ClusterRestService.getNodeStatus()
-            .success((data) => {
-                $scope.currentNode = data;
+            .then((response) => {
+                $scope.currentNode = response.data;
             })
-            .error((data) => {
-                $scope.currentNode = data;
+            .catch((error) => {
+                $scope.currentNode = error.data;
                 $scope.clusterModel.hasCluster = false;
             })
-            .finally(() => getLocations().then(() => getRemoteLocationsRpcAddresses($scope.clusterModel.locations)));
+            .then(() => getLocationsWithRpcAddresses());
     };
 
-    function getLocations() {
-        return LocationsRestService.getLocations()
-            .success(function (response) {
-                const localNode = response.find((location) => location.local);
-                localNode.uri = $scope.currentNode.endpoint;
+    function getLocationsWithRpcAddresses() {
+        return RemoteLocationsService.getLocationsWithRpcAddresses()
+            .then((locations) => {
+                const localNode = locations.find((location) => location.isLocal);
+                localNode.endpoint = $scope.currentNode.endpoint;
```

Comment:
`localNode` can be undefined here. Are you sure you don't need a check?

Reply:
Added a NPE check